resharper/src/resharper-unity/CSharp/Feature/Services/QuickFixes/PreferAddressByIdToGraphicsParamsQuickFix.cs

```diff
@@ -0,0 +1,258 @@
+using System;
+using JetBrains.Annotations;
+using JetBrains.Application.Progress;
+using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Feature.Services.QuickFixes;
+using JetBrains.ReSharper.Plugins.Unity.CSharp.Daemon.Errors;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.CSharp;
+using JetBrains.ReSharper.Psi.CSharp.Tree;
+using JetBrains.ReSharper.Psi.CSharp.Util;
+using JetBrains.ReSharper.Psi.ExtensionsAPI.Resolve.Managed;
+using JetBrains.ReSharper.Psi.Naming.Extentions;
+using JetBrains.ReSharper.Psi.Naming.Impl;
+using JetBrains.ReSharper.Psi.Naming.Settings;
+using JetBrains.ReSharper.Psi.Resolve.Managed;
+using JetBrains.ReSharper.Psi.Tree;
+using JetBrains.TextControl;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.Unity.CSharp.Feature.Services.QuickFixes
+{
+    [QuickFix]
+    public class PreferAddressByIdToGraphicsParamsQuickFix : QuickFixBase
+    {
+        private readonly IInvocationExpression myInvocationExpression;
+        private readonly ICSharpArgument myArgument;
+        
+        // using for name of new generated field
+        private readonly string myFieldName;
+        
+        // using for find already declared field or property
+        private readonly string myGraphicsPropertyName;
+
+        // using to handle concatenation operations with const values and save possibility to correct refactoring
+        private readonly IExpression myArgumentExpression;
+        private readonly string myMapFunction;
+        private readonly string myTypeName; 
+        
+        public PreferAddressByIdToGraphicsParamsQuickFix(PreferAddressByIdToGraphicsParamsWarning warning)
+        {
+            myInvocationExpression = warning.InvocationMethod;
+            myArgument = warning.Argument;
+            myFieldName = myGraphicsPropertyName = warning.Literal;
+            if (!ValidityChecker.IsValidIdentifier(myFieldName))
+            {
+                myFieldName = "Property";
+            }
+
+            myArgumentExpression = warning.ArgumentExpression;
+            myMapFunction = warning.MapFunction;
+            myTypeName = warning.TypeName;
+        }
+        
+        protected override Action<ITextControl> ExecutePsiTransaction(ISolution solution, IProgressIndicator progress)
+        {
+            var factory = CSharpElementFactory.GetInstance(myInvocationExpression);
+
+            //  try find declaration where string to id conversation is done. If we don't find it, create in top-level class
+            var idDeclaration = TryFindDeclaration(myInvocationExpression, myGraphicsPropertyName, myTypeName, myMapFunction, out var name); 
```

Comment:
Looks like `out var name` is always overwritten before usage. Can we get rid of it?